### ISSUE TITLE
document netCDF logging

### ISSUE THREAD

**Issue by edwardhartnett** [Unidata/netcdf-c#2676]
Several advanced users at NOAA would benefit from documentation of the netCDF-4 logging capability.

This documentation must contain sufficient provisos and cautions, so the user understands that logging output is not part of the API and can be changed at any time.

However, with appropriate cautions, the logging capability can help debug really hard to catch bugs, especially in parallel I/O situations, where the bug might only manifest when scales are increased. Debugging software on hundreds or thousands of processors is challenging.
